src/chainstate/stacks/db/blocks.rs

```diff
@@ -1998,6 +2016,8 @@ impl StacksChainState {
         }
     }
 
+    /*
+    // NOTE: temporarily not used, so we can kee invalid data around for analysis.
```

Comment:
Would prefer to just delete rather than comment out -- the code will be stored in the git history in any case.

Reply:
Actually -- would prefer if this path were updated like in #2501 (https://github.com/blockstack/stacks-blockchain/pull/2501/files#diff-36b251011e9d6834be636547f23dd28aa9b24b555331f637dd40cfd2a1552467R2018) --- the microblock data is still removed from the table, but it is inserted into a new invalid table as well.

Reply:
Copying invalid microblocks -- i.e. ones that could be invalid because they take up a lot of space -- to a different table isn't exactly a free operation.  I'm not sure what we'd be getting out of it?  IIRC sqlite doesn't do online defragmentation, so unless we explicitly `VACUUM` the database, deleting the blobs isn't going to improve query speed on the existing microblocks.  Putting them into a separate table doesn't really save us anything on query complexity, because you're just replacing a `WHERE orphaned = 1` with a different table name (so you're choosing between having a separate table index, or an index on `orphaned`).

Reply:
I'm pretty sure these codepaths only try to read microblocks where `orphaned = 0` already, but if not, then they should do so regardless of which way this goes (because an orphaned microblock can remain resident in the DB in-between two or more calls to `process_next_staging_block()`).

Reply:
Right -- I agree that other codepaths _should_ only try to read where `orphaned = 0`, but it's a safer practice to make the more "pessimistic" change here.